---
title: "community analyses"
author: "Max based on Roey"
date: "march 2019"
output: html_document
---


## the dataset has been modified, and normalized

## set working directory
```{r}
# setwd("C:/Users/Max/Desktop/DoME-Data/MiSeq/MiSeq_fierer_grasslands/MiSeq_combined_Nepel_fierer_grasslands_220716_Thomas/dataSetList.txt.nifH.0.03.OTU")
setwd("\\\\share4.univie.ac.at/DOME/staff/nepel/MiSeq/Fierer_grasslands_MiSeq/MiSeq_combined_Nepel_fierer_grasslands_220716_Thomas/fierer_foR_manuscript/")
# setwd("Y:/staff/nepel/MiSeq/Fierer_grasslands_MiSeq/MiSeq_combined_Nepel_fierer_grasslands_220716_Thomas/fierer_foR_manuscript/")
```

# libraries
```{r}
library(vegan)
library(magrittr)
library(phyloseq)
library(ggplot2)
# library(devtools)
# install_github("microbiome/microbiome")
# library(microbiome) # meta()
library(reshape2) #melt()
library(plyr) 
```

# functions
... used, hidden ...
```{r, echo=FALSE}
PlotOrd <- function(data2plot, components = c("PC1", "PC2"), groups = groups, treatment = treatment, variance = TRUE, connect = FALSE, path = groups, species = NULL, ...){
  ## plot ordinations as a biplot 
  # Arguments: 
  # data - a data frame containing sample and/or species scores from an ordination analysis
  # species - an optional species scores to be plotted as vectors or points
  # variance <-   print variance explained to the axis titles
  # axes <- choose axes to plot
  #   two.groups <- c("#006837", "#1A9850", "#A6D96A", "#D9EF8B", "#A50026", "#D73027", "#FDAE61", "#FEE08B") # based on brewer: RdYlGn
  p <- ggplot(data2plot, aes_string(x = components[1], y = components[2])) +
    geom_point(aes_string(colour = groups, shape = treatment), size = 5, alpha = 1) +
    #     scale_size_discrete(range = c(8,8)) +
    theme(panel.background = element_rect(fill = "#F2F2F2"), 
          panel.grid.minor = element_blank()) + 
    guides(colour = guide_legend(title = ""), shape = guide_legend(title = "")) +
#  stat_ellipse(type = "t") +
#     scale_colour_manual(values = NevadaRainbow) +
#     scale_colour_brewer(palette = "Dark2") # w/o this command -> unlimited standard colours
#      scale_colour_brewer(palette = "Paired") # testing!
  #     theme(legend.position = c(0.9, 0.76), legend.background = element_rect(fill = "white", colour = NA))
  if (connect == TRUE) {
    p <- p + geom_path(aes_string(group = path, colour = groups), alpha = 1/2) + guides(colour = guide_legend(title = "Gradient"))
  }
  if (!is.null(species)) {
    p + geom_path(data2plot = arrws, aes(PC1, PC2, group = species), 
                  colour = "red", arrow = arrow(length = unit(0.05, "npc"))) + 
      # scale_colour_gradient(limits = c(1,4) legend = F) +
      geom_text(data = arrws[5:8, ], aes(label = species), hjust = 0.5, vjust = -1, colour = "black")
  }
  return(p)
}

TaxThresh <- function(Taxonomy, tax.level = "Order", thresh = 1){
  ## clump together all taxa with frequency below threshold as "rare" and all those with bs values below threshold as "unclassified"
  # vars: Taxonomy=Taxonomy df, bs.vals=bs values df, bs.thresh=threshold of bs value to drop, tax.level=which level should be considered, thresh=threshold for "rare"
  # make all below bs threshold unclassified
  Taxonomy[, tax.level] <- factor(Taxonomy[, tax.level], levels = c(levels(Taxonomy[, tax.level]), "Rare")) # add "Rare" and "Unclassified" levels

  #   Taxonomy[bs.vals[, tax.level] < bs.thresh,tax.level] <- factor("Unclassified") # replace in Taxonomy all rows under tax.level whose corresponding bs.vals are below bs.thresh with "unclassified"

  # count frequencies of taxonomies in samples (unique sequences only * their counts = all sequences)
  count.tax <- plyr::count(df = Taxonomy, vars = tax.level, wt_var = "Freq") # data.frame, Vars, weight
  count.tax[, 2] <- (count.tax[, 2] / sum((count.tax[, 2]))) * 100 # norm to 100%

  ab.tax <- droplevels(count.tax[count.tax$freq > thresh, ]) # remove all taxa below threshold (keep abundent taxa)

  # generate logical vector of abundant taxa
  inds <- vector(mode = "logical", length = nrow(Taxonomy)) # assign a logical vector length Taxonomy
  # cycle through taxa and generate logical vector of abundant taxa
  for (i in 1:length(levels(ab.tax[, 1]))) inds <- as.logical(inds + grepl(levels(ab.tax[,1])[i], Taxonomy[, tax.level]))
  possibleError <- tryCatch(Taxonomy[!inds, tax.level] <- "Rare", error = function(e) e) # replace all rare taxa with "Rare"
  ab.Taxonomy <- droplevels(Taxonomy) # remove factor levels reclassified as either "rare" or "unclassified"
  return(ab.Taxonomy)
}

barPalette4 <- colorRamps::primary.colors(n=4)
barPalette15 <- colorRamps::primary.colors(n=15)
barPalette17 <- colorRamps::primary.colors(n=17)
barPalette18 <- colorRamps::primary.colors(n=18)
barPalette19 <- colorRamps::primary.colors(n=19)
barPalette20 <- colorRamps::primary.colors(n=20)
barPalette25 <- colorRamps::primary.colors(n=25)
barPalette28 <- colorRamps::primary.colors(n=28)
barPalette31 <- colorRamps::primary.colors(n=31)
barPalette32 <- colorRamps::primary.colors(n=32)
barPalette42 <- colorRamps::primary.colors(n=42)
barPalette46 <- colorRamps::primary.colors(n=46)
barPalette71 <- colorRamps::primary.colors(n=71)
```

# loading modified and normalized datasets
```{r}
load(file = "prep.otuCountTable/normalizations.Robj")
```

# defining normalized dataset to use and create phyloseq pbject
```{r}
OTUs <- OTU.mat.rare # .rel, .norm, .cut
labels <- labels.rare
tax <- taxonomy.rare

# creating phyloseq object of the selected normalized dataset
otu.physq <- otu_table(OTUs, taxa_are_rows=FALSE)
sam.physq <- sample_data(labels) 
tax.physq <- tax_table(tax)
(physeq.gras <- phyloseq(otu.physq, sam.physq, tax.physq))

# creating phyloseq object for non-rarefied dataset
otu.physq <- otu_table(OTU.mat.cut, taxa_are_rows=FALSE)
sam.physq <- sample_data(labels.cut) 
tax.physq <- tax_table(taxonomy.cut)
(physeq.gras.absolut <- phyloseq(otu.physq, sam.physq, tax.physq))
```

# separate numeric from non-numeric factors
```{r}
labels.2split <- labels
write.table(labels.2split, paste("fierer.analyses/txt_saves/labels",Sys.Date(),"txt",sep = "."), sep="\t", row.names=FALSE)

factors <- colnames(labels.2split)
factors
list <-  as.list(colnames(labels.2split))
class(factors)

cat <- c("conc", "site_code", "cover_year", "site_name", "continent", "country", "region", "managed", "burned", "grazed", "anthropogenic", "habitat", "N", "P", "trt_yn", "Exclose", "year_trt", "trt", "nutrient_trt", "Classification", "LCvsHC", "site_trt", "site_nutrient_trt", "trt_N", "trt_P", "trt_NP", "trt_NNP", "trt_NPP")
labels.cat <- labels.2split[cat]
# labels.cat <- labels.2split[1:nrow(labels.2split),c(3:4, 7:11, 17, 48:49, 90, 92:93)] # only relevante categorical factors


num <- c("elevation", "latitude", "longitude", "RAIN_PET", "MAT", "MAT_RANGE", "ISO", "TEMP_VAR", "MAX_TEMP", "MIN_TEMP", "ANN_TEMP_RANGE", "TEMP_WET_Q", "TEMP_DRY_Q", "TEMP_WARM_Q", "TEMP_COLD_Q", "MAP", "MAP_WET_M", "MAP_DRY_M", "MAP_VAR", "MAP_WET_Q", "MAP_DRY_Q", "MAP_WARM_Q", "MAP_COLD_Q", "AI", "PET", "sum_INT_cover", "sum_NAT_cover", "sum_UNK_cover", "total_cover", "INT_rich", "NAT_rich", "UNK_rich", "rich", "site_year_rich", "site_richness", "site_native_richness", "site_introduced_richness", "plot_beta", "dead_mass", "live_mass", "total_mass", "proportion_par", "Ambient_PAR", "Ground_PAR", "soil_pctC", "soil_pctN", "soil_ppmP", "soil_ppmK", "soil_pH", "PercentSand", "PercentSilt", "PercentClay", "rich.vegan", 
"shan", "simpson", "even")

labels.2split[num] %>%
  as.matrix() %>%
  apply(., 2, function(x) as.numeric(x)) -> labels.num
  rownames(labels.num) <- rownames(labels.2split)
  
# labels.2split[1:nrow(labels.2split),c(17:22, 24:39, 67:68, 72:78, 83:86)] %>% # only relevante numerical factors  
#     as.matrix() %>%
#     apply(., 2, function(x) as.numeric(x)) -> labels.num #36 warnings = neg.controls
#   rownames(labels.num) <- rownames(labels.2split)

labels.f <- labels.2split[1:nrow(labels.2split),c(17:22, 24:39, 67:68, 72:78, 83:86)]
# write.table(labels.cat, paste("fierer.analyses/txt_saves/labels.cat",Sys.Date(),"txt",sep = "."), sep="\t", row.names=TRUE)
# write.table(labels.num, paste("fierer.analyses/txt_saves/labels.num",Sys.Date(),"txt",sep = "."), sep="\t", row.names=TRUE)
# write.table(labels.f, paste("fierer.analyses/txt_saves/labels.f",Sys.Date(),"txt",sep = "."), sep="\t", row.names=TRUE)
```

####################################################################################################
###################################### categorical variables #######################################
############################################# adonis ###############################################

# testing if fence/controls are significant different
```{r}
OTUs.cont.0 <- OTUs[grep("control", labels$trt_yn),]
OTUs.cont <- OTUs.cont.0[,!colSums(OTUs.cont.0) == 0 ] # exclude OTUs, which have in sum 0 reads
labels.cont <- labels[grep("control", labels$trt_yn),]
tax.cont <- tax[!rowSums(t(OTUs.cont.0)) == 0, ]

(mod.cont <- adonis(OTUs.cont ~ site_code * trt, data = labels.cont, method = "bray", permutations = 999))
# -> no significant difference between fenced and normal controls within each site
```

# Model categorical using Adonis
```{r}
(mod.site <- adonis(OTUs ~ site_code, data = labels, method = "bray", permutations = 999))
(mod.1 <- adonis(OTUs ~ site_code * trt_NNP * trt_NPP, data = labels, method = "bray", permutations = 999))
# (mod.2 <- adonis(OTUs ~ site_code + trt_NNP + trt_NPP + site_code:trt_NNP, data = labels, method = "bray", permutations = 999))
# (mod.4 <- adonis(OTUs ~ continent * country * region * site_code * trt_NNP * trt_NPP, data = labels, method = "bray", permutations = 999))
(mod.3 <- adonis(OTUs ~ trt_NNP * trt_NPP, strata = labels$site_code, data = labels, method = "bray", permutations = 999))
```

## phyloseq - subsetting sites
```{r}
#######################all sites!! - working####################

site.list <- levels(labels$site_code)
# site.list <- site.list[-14] #exclude mtca.au

d.subs<-list()
for(i in site.list)
{
 d.subs[[i]] <- subset_samples(physeq.gras, site_code%in% i)
 assign(paste("physeq.trt.", i, sep = ""), d.subs[[i]])
}

# creating a list of physeq objects of all sites
physeq.sites0 <- data.frame(matrix(ncol=1,nrow=0))
for(i in site.list)
{
physeq.sites0[i,1] <- paste("physeq.trt.", i, sep = "")
}
physeq.sites0.list <- physeq.sites0[,1]


otu.trt.bnch.us <- otu_table(physeq.trt.bnch.us)
meta.trt.bnch.us <- sample_data(physeq.trt.bnch.us) 

otu.trt.burrawan.au <- otu_table(physeq.trt.burrawan.au)
meta.trt.burrawan.au <- sample_data(physeq.trt.burrawan.au) 

otu.trt.cbgb.us <- otu_table(physeq.trt.cbgb.us)
meta.trt.cbgb.us <- sample_data(physeq.trt.cbgb.us) 

otu.trt.cdpt.us <- otu_table(physeq.trt.cdpt.us)
meta.trt.cdpt.us <- sample_data(physeq.trt.cdpt.us) 

otu.trt.cowi.ca <- otu_table(physeq.trt.cowi.ca)
meta.trt.cowi.ca <- sample_data(physeq.trt.cowi.ca) 

otu.trt.elliot.us <- otu_table(physeq.trt.elliot.us)
meta.trt.elliot.us <- sample_data(physeq.trt.elliot.us) 

otu.trt.frue.ch <- otu_table(physeq.trt.frue.ch)
meta.trt.frue.ch <- sample_data(physeq.trt.frue.ch) 

otu.trt.gilb.za <- otu_table(physeq.trt.gilb.za)
meta.trt.gilb.za <- sample_data(physeq.trt.gilb.za) 

otu.trt.hall.us <- otu_table(physeq.trt.hall.us)
meta.trt.hall.us <- sample_data(physeq.trt.hall.us) 

otu.trt.hart.us <- otu_table(physeq.trt.hart.us)
meta.trt.hart.us <- sample_data(physeq.trt.hart.us) 

otu.trt.konz.us <- otu_table(physeq.trt.konz.us)
meta.trt.konz.us <- sample_data(physeq.trt.konz.us) 

otu.trt.lancaster.uk <- otu_table(physeq.trt.lancaster.uk)
meta.trt.lancaster.uk <- sample_data(physeq.trt.lancaster.uk) 

otu.trt.look.us <- otu_table(physeq.trt.look.us)
meta.trt.look.us <- sample_data(physeq.trt.look.us) 

otu.trt.mtca.au <- otu_table(physeq.trt.mtca.au)
meta.trt.mtca.au <- sample_data(physeq.trt.mtca.au)

otu.trt.sage.us <- otu_table(physeq.trt.sage.us)
meta.trt.sage.us <- sample_data(physeq.trt.sage.us) 

otu.trt.saline.us <- otu_table(physeq.trt.saline.us)
meta.trt.saline.us <- sample_data(physeq.trt.saline.us) 

otu.trt.shps.us <- otu_table(physeq.trt.shps.us)
meta.trt.shps.us <- sample_data(physeq.trt.shps.us) 

otu.trt.sier.us <- otu_table(physeq.trt.sier.us)
meta.trt.sier.us <- sample_data(physeq.trt.sier.us) 

otu.trt.smith.us <- otu_table(physeq.trt.smith.us)
meta.trt.smith.us <- sample_data(physeq.trt.smith.us) 

otu.trt.spin.us <- otu_table(physeq.trt.spin.us)
meta.trt.spin.us <- sample_data(physeq.trt.spin.us) 

otu.trt.summ.za <- otu_table(physeq.trt.summ.za)
meta.trt.summ.za <- sample_data(physeq.trt.summ.za) 

otu.trt.ukul.za <- otu_table(physeq.trt.ukul.za)
meta.trt.ukul.za <- sample_data(physeq.trt.ukul.za) 

otu.trt.unc.us <- otu_table(physeq.trt.unc.us)
meta.trt.unc.us <- sample_data(physeq.trt.unc.us) 

otu.trt.valm.ch <- otu_table(physeq.trt.valm.ch)
meta.trt.valm.ch <- sample_data(physeq.trt.valm.ch)

```

## adonis NNP & NPP per site
```{r}
order <- c("bnch.us","burrawan.au","cbgb.us","cdpt.us","cowi.ca","elliot.us","frue.ch","gilb.za","hall.us","hart.us","konz.us","lancaster.uk","look.us","sage.us","saline.us","shps.us","sier.us","smith.us","spin.us","summ.za","ukul.za","unc.us","valm.ch")
otu.list <- paste0("otu.trt", ".", order)
meta.list1 <- paste0("meta.trt", ".", order, "$trt_NNP")
meta.list2 <- paste0("meta.trt", ".", order, "$trt_NPP")


pvs <- data.frame(matrix(ncol=4,nrow=23))
row.names(pvs) <- order
pvs$site_code  <- order
colnames(pvs) <- c("R2.NNP", "p.value.NNP", "R2.NPP", "p.value.NPP")#, "site_code")

for (i in 1:23) { 
    form <- as.formula(paste(otu.list[i], paste(meta.list1[i], meta.list2[i], sep="*"), sep="~"))
    a <- (adonis(form, method = "bray", permutations=999,na.rm = TRUE))
    pvs[i,1]<- a$aov.tab$R2[1]
    pvs[i,2]<- a$aov.tab$Pr[1]
    pvs[i,3]<- a$aov.tab$R2[2]
    pvs[i,4]<- a$aov.tab$Pr[2]
}
pvs$p.value.adjust.NNP <- p.adjust(pvs$p.value.NNP, method="BH") #"holm", "hochberg", "hommel", "bonferroni", "BH", "BY"
pvs$p.value.adjust.NPP <- p.adjust(pvs$p.value.NPP, method="BH") #"holm", "hochberg", "hommel", "bonferroni", "BH", "BY"
print(pvs)
write.table(pvs, paste("fierer.analyses/txt_saves/adonis.per.site",Sys.Date(),"txt",sep = "."), sep="\t", row.names=FALSE)
```

####################################################################################################
###################################         all data            ####################################
###################################      alpha diversity        ####################################
####################################################################################################

# Alpha diversität - phyloseq - not rarefy beforehand, it is taken into account by plot_richness()
```{r}
(physeq.gras.absolut)
## rarefied alpha div <- not the right way
# plot_richness(physeq.gras.absolut, measures = c("Chao1", "Shannon", "Simpson"))
# plot_richness(physeq.gras.absolut, measures = c("Chao1"))
plot_richness(physeq.gras, measures = c("Shannon"))
```

####################################################################################################
###################################         all data            ####################################
###################################        ordinations          ####################################
####################################################################################################

# select data to plot
```{r}
OTU2plot <- OTUs
Labels <- labels
plottitle <- "PCoA rarefied to 500 reads"

# OTU2plot <- OTU2plot[-grep("Europe", Labels$continent),]
# Labels <- labels[-grep("Europe", Labels$continent),]

# OTU2plot <- OTU2plot[-grep("Africa", Labels$continent),]
# Labels <- labels[-grep("Africa", Labels$continent),]

# OTU2plot <- OTU2plot[grep("North_America", Labels$continent),]
# Labels <- labels[grep("North_America", Labels$continent),]
```

# Make UNconstrained ordination (PCoA)
```{r}
pcoa.obj <- capscale((OTU2plot) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn)

colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "site_code", treatment = "continent", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  # stat_ellipse(type = "t") +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))

ggsave(paste("fierer.analyses/graphs/plot.pcoa",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.pcoa",Sys.Date(),"pdf",sep = "."),width = 10, height = 8, dpi = 300)
```

## PCoA including ellipses!
```{r}
ord <- ordinate(physeq.gras, formula = ~1, "CAP", "bray")
(ordplot <- plot_ordination(physeq.gras, ord, "sampleID", color="site_code",shape="continent"))

ordplot + 
  stat_ellipse(type = "norm", linetype = 2) + 
  geom_point(size = 4) +
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))
  # stat_ellipse(type = "t") +
  # theme_bw()

ggsave(paste("fierer.analyses/graphs/plot.pcoa.ell",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.pcoa.ell",Sys.Date(),"pdf",sep = "."),width = 10, height = 8, dpi = 300)
```

# Make constrained ordination (dbRDA) using significant/ important factors
## ~trt_NNP
```{r}
plottitle <- "dbRDA ~ trt_NNP; rarefied to 500 reads"
dbRDA.obj <- capscale((OTU2plot) ~ trt_NNP, data = Labels, distance = "bray") # horn, bray # ~ live_mass, #OTU2plot #Labels

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP = Labels$trt_NNP, trt_NPP = Labels$trt_NPP)
colnames(dbRDA.df)[c(2,3)] <- c("CAP1", "PC2")
p <- PlotOrd(dbRDA.df , components = c("CAP1", "PC2"), groups = "trt_NNP", treatment = "continent", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP",Sys.Date(),"pdf",sep = "."),width = 10, height = 8, dpi = 300)
```

####################################################################################################
###################################           all data           ###################################
###################################      taxonomy bar plots      ###################################
####################################################################################################

### base for stacked bar charts & heatmaps @ taxonomic level
```{r}
OTU4tax <- as.data.frame(t(OTUs))
  OTU4tax$Total <- rowSums(OTU4tax)
  OTU4tax$OTU <- rownames(OTU4tax)

meta <- labels#[,c(4,8:11,16:20, 98)] # one entry per site_code (=24) # ncol(labels.rare)
meta$Name <- row.names(meta)
dim(meta)
tax4otu <- tax
dim(tax4otu)

Taxonomy <- cbind(tax4otu, OTU4tax) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)
Taxonomy <- melt(Taxonomy, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

# ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.2) # get ONLY abundant taxonomy
# ab.Taxonomy.class <- merge(ab.Taxonomy.class, meta, by="Name") # all 316 sites

Taxonomy <- merge(Taxonomy, meta, by="Name")

ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.genus <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.z.cluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Cluster", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.z.subcluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Subcluster", thresh = 0.2) # get ONLY abundant taxonomy
```

#### --> stacked bar charts & heatmap @ Class level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.class
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Zehr.Cluster = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und Sample # wurde ev.schon durch merge_samples() in phyloseq gemacht

  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$sam.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $site_code
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$site_code == tax.data2plot$site_code[i]])
  tax.data2plot$site.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per Sample
sam.rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=sam.Rel.Ab,fill=Taxa, main="rel. ab. class_id level per site") +
  scale_fill_manual(values=barPalette20) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
sam.rel.bar  + facet_wrap(~site_code, scale="free", ncol=4) #

ggsave(paste("fierer.analyses/graphs/bar.all.class.fw.site",Sys.Date(),"png",sep = "."),width = 20, height = 15, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.all.class.fw.site",Sys.Date(),"pdf",sep = "."),width = 20, height = 15, dpi = 300)

  ## plot stacked bar chart per site_code
site.rel.bar <- qplot(x=site_code,data=tax.data2plot,geom="bar",weight=site.Rel.Ab,fill=Taxa, main="rel. ab. class level per site") +
  scale_fill_manual(values=barPalette20) +
  scale_x_discrete(limits=sort4) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
site.rel.bar # + facet_wrap(~continent, ncol=1, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.all.class.perSite",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.all.class.perSite",Sys.Date(),"pdf",sep = "."),width = 10, height = 6, dpi = 300)
```

#### --> stacked bar charts & heatmap @ Zehr.Cluster Level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.z.cluster
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Class = ab.Taxonomy$Class, Taxa = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und Sample # wurde ev.schon durch merge_samples() in phyloseq gemacht

  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$sam.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $site_code
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$site_code == tax.data2plot$site_code[i]])
  tax.data2plot$site.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per Sample
sam.rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=sam.Rel.Ab,fill=Taxa, main="rel. ab. class_id level per site") +
  scale_fill_manual(values=barPalette20) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
sam.rel.bar  + facet_wrap(~site_code, scale="free", ncol=4) #

ggsave(paste("fierer.analyses/graphs/bar.all.z.cl.fw.site",Sys.Date(),"png",sep = "."),width = 20, height = 15, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.all.z.cl.fw.site",Sys.Date(),"pdf",sep = "."),width = 20, height = 15, dpi = 300)

  ## plot stacked bar chart per site_code
site.rel.bar <- qplot(x=site_code,data=tax.data2plot,geom="bar",weight=site.Rel.Ab,fill=Taxa, main="rel. ab. class level per site") +
  scale_fill_manual(values=barPalette20) +
  scale_x_discrete(limits=sort4) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
site.rel.bar # + facet_wrap(~continent, ncol=1, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.all.z.cl.perSite",Sys.Date(),"png",sep = "."),width = 10, height = 6, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.all.z.cl.perSite",Sys.Date(),"pdf",sep = "."),width = 10, height = 6, dpi = 300)
```


####################################################################################################
###################################        ordinations           ###################################
################################### ONLY 2 NNP significant sites ###################################
####################################################################################################

# prepare data to plot ONLY look.us + konz.us
```{r}
# ## add colums 0/1 for the 3 significant sites (adonis trt_NNP results)
#   labels$sign <- 0
#     levels(labels$sign) <- c(levels(labels$sign), 0, 1)
#       labels$sign[labels$site_code == "look.us"] <- 1
#       labels$sign[labels$site_code == "konz.us"] <- 1
#       # Labels$sign[Labels$site_code == "sier.us"] <- 1
#       labels$sign[!labels$sign == "1"] <- 0
# 
# OTU2plot <- OTUs[grep("1", labels$sign),]
# Labels <- labels[grep("1", labels$sign),]
```

# Make UNconstrained ordination (PCoA) for trt_NNP significant sites
##  - look.us
```{r}
OTU2plot <- OTUs[grep("look.us", labels$site_code),] # if only for one specific - look.us, konz.us
Labels <- labels[grep("look.us", labels$site_code),] # if only for one specific - look.us, konz.us
plottitle <- "PCoA trt_NNP sign. site - look.us; rarefied to 500 reads"

pcoa.obj <- capscale((OTU2plot) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP=Labels$trt_NNP)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "trt_NNP", treatment = "site_code", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) + 
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))  #+
  # facet_wrap(~ site_code, scale="free")

# ggsave("pcoa_unconstrained_look5.png",width = 5, height = 4, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.PCoA.look.us",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.PCoA.look.us",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

##  - konz.us
```{r}
OTU2plot <- OTUs[grep("konz.us", labels$site_code),] # if only for one specific - look.us, konz.us
Labels <- labels[grep("konz.us", labels$site_code),] # if only for one specific - look.us, konz.us
plottitle <- "PCoA trt_NNP sign. site - konz.us; rarefied to 500 reads"

pcoa.obj <- capscale((OTU2plot) ~ 1, distance = "bray")

dbRDA.scores <- scores(pcoa.obj)
explained <- eigenvals( pcoa.obj )/sum( eigenvals( pcoa.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP=Labels$trt_NNP)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")

p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "trt_NNP", treatment = "site_code", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) + 
  xlab(paste("PC 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("PC 2 (", explained[2], "%)", sep = ""))  #+
  # facet_wrap(~ site_code, scale="free")

ggsave(paste("fierer.analyses/graphs/plot.PCoA.konz.us",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.PCoA.konz.us",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

# Make constrained ordination (dbRDA) using significant/ important factors  -  konz.us + look.us
##  - konz.us ~ trt_NNP
```{r}
OTU2plot <- OTUs[grep("konz.us", labels$site_code),] # if only for one specific - look.us, konz.us
Labels <- labels[grep("konz.us", labels$site_code),] # if only for one specific - look.us, konz.us
plottitle <- "dbRDA ~ trt_NNP - konz.us; rarefied to 500 reads"

dbRDA.obj <- capscale((OTU2plot) ~ trt_NNP, data = Labels, distance = "bray") # horn, bray # ~ live_mass, #OTU2plot #Labels

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP = Labels$trt_NNP, trt_NPP = Labels$trt_NPP)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "trt_NNP", treatment = "site_code", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = "")) #+
  # facet_wrap(~ site_code, scale="free")

ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP.konz.us",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP.konz.us",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

##  - look.us ~ trt_NNP
```{r}
OTU2plot <- OTUs[grep("look.us", labels$site_code),] # if only for one specific - look.us, konz.us
Labels <- labels[grep("look.us", labels$site_code),] # if only for one specific - look.us, konz.us
plottitle <- "dbRDA ~ trt_NNP - look.us; rarefied to 500 reads"

dbRDA.obj <- capscale((OTU2plot) ~ trt_NNP, data = Labels, distance = "bray") # horn, bray # ~ live_mass, #OTU2plot #Labels

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP = Labels$trt_NNP, trt_NPP = Labels$trt_NPP)
colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "trt_NNP", treatment = "site_code", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = "")) #+
  # facet_wrap(~ site_code, scale="free")

ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP.look.us",Sys.Date(),"png",sep = "."),width = 7, height = 5, dpi = 300)
ggsave(paste("fierer.analyses/graphs/plot.dbRDA.trt_NNP.look.us",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

################################ taxonomy graphs - ONLY significant sites ################################
############################################ look.us & konz.us ###########################################

## phyloseq - preparing data for modelling & plotting
```{r}
# only significant N-treatment sites "look.us", "konz.us"
(physeq.sign <- subset_samples(physeq.gras, site_code%in%(c("look.us", "konz.us"))))
  (physeq.sign <- prune_taxa(taxa_sums(physeq.sign) > 0, physeq.sign))

otu.sign <- as.data.frame(t(otu_table(physeq.sign)))
  otu.sign$Total <- rowSums(otu.sign)
  otu.sign$OTU <- rownames(otu.sign)

tax.sign <- as.data.frame(tax_table(physeq.sign))
meta.sign <- meta(physeq.sign) # not useful
  meta.sign$Name <- row.names(meta.sign) # for merge() next chunk
```

### base for stacked bar charts & heatmaps @ taxonomic level
```{r}
Taxonomy <- cbind(tax.sign, otu.sign) # combine data.frames (OTU/Taxonomy & Sample Abundance; merge doesnt work on such large dbs)

Taxonomy <- melt(Taxonomy, id = c("OTU", "Total", "Domain", "Phylum", "Class", "Order", "Family", "Genus", "Zehr.Cluster", "Zehr.Subcluster"), variable.name = 'Name', value.name = 'Freq', na.rm = TRUE) # --> jede OTU in jedem sample einen eintrag mit frequenz

Taxonomy <- merge(Taxonomy, meta.sign, by="Name")

ab.Taxonomy.class <- TaxThresh(Taxonomy, tax.level = "Class", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.order <- TaxThresh(Taxonomy, tax.level = "Order", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.family <- TaxThresh(Taxonomy, tax.level = "Family", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.genus <- TaxThresh(Taxonomy, tax.level = "Genus", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.z.cluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Cluster", thresh = 0.2) # get ONLY abundant taxonomy
ab.Taxonomy.z.subcluster <- TaxThresh(Taxonomy, tax.level = "Zehr.Subcluster", thresh = 0.2) # get ONLY abundant taxonomy
```

#### --> stacked bar charts & heatmap @ Class level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.class
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Class, Zehr.Cluster = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und Sample # wurde ev.schon durch merge_samples() in phyloseq gemacht

# for stacked bar charts
  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $trt_NNP
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$trt_NNP == tax.data2plot$trt_NNP[i]])
  tax.data2plot$NNP.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per NNP
rel.bar <- qplot(x=trt_NNP,data=tax.data2plot,geom="bar",weight=NNP.Rel.Ab,fill=Taxa, main="rel. ab. Class level per site") + scale_fill_manual(values=barPalette20) #rel. Abundance
rel.bar + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

  ## plot stacked bar chart per SAMPLE
rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa, main="rel. ab. Class level per site") +
  scale_fill_manual(values=barPalette20) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
rel.bar  + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.sign.class.fw.siteNNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.sign.class.fw.siteNNP",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

#### --> stacked bar charts & heatmap @ Order level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.order
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Order, Class = ab.Taxonomy$Class, Zehr.Cluster = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und Sample # wurde ev.schon durch merge_samples() in phyloseq gemacht

# for stacked bar charts
  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $trt_NNP
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$trt_NNP == tax.data2plot$trt_NNP[i]])
  tax.data2plot$NNP.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per NNP
rel.bar <- qplot(x=trt_NNP,data=tax.data2plot,geom="bar",weight=NNP.Rel.Ab,fill=Taxa, main="rel. ab. Order level per site") + scale_fill_manual(values=barPalette28) #rel. Abundance
rel.bar + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

  ## plot stacked bar chart per SAMPLE
rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa, main="rel. ab. Order level per site") +
  scale_fill_manual(values=barPalette28) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
rel.bar  + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.sign.order.fw.siteNNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.sign.order.fw.siteNNP",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

#### --> stacked bar charts & heatmap @ Family level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.family
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Family, Class = ab.Taxonomy$Class, Order = ab.Taxonomy$Order, Zehr.Cluster = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und Sample # wurde ev.schon durch merge_samples() in phyloseq gemacht

# for stacked bar charts
  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $trt_NNP
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$trt_NNP == tax.data2plot$trt_NNP[i]])
  tax.data2plot$NNP.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per NNP
rel.bar <- qplot(x=trt_NNP,data=tax.data2plot,geom="bar",weight=NNP.Rel.Ab,fill=Taxa, main="rel. ab. Order level per site") + scale_fill_manual(values=barPalette46) #rel. Abundance
rel.bar + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

  ## plot stacked bar chart per SAMPLE
rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa, main="rel. ab. Order level per site") +
  scale_fill_manual(values=barPalette28) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
rel.bar  + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.sign.order.fw.siteNNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.sign.order.fw.siteNNP",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

#### --> stacked bar charts & heatmap @ Zehr.Cluster level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.z.cluster
#Zehr.Cluster
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Zehr.Cluster, Zehr.Subcluster = ab.Taxonomy$Zehr.Subcluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und site_code # wurde ev.schon durch merge_samples() in phyloseq gemacht

# for stacked bar charts
  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $trt_NNP
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$trt_NNP == tax.data2plot$trt_NNP[i]])
  tax.data2plot$NNP.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per NNP
rel.bar <- qplot(x=trt_NNP,data=tax.data2plot,geom="bar",weight=NNP.Rel.Ab,fill=Taxa, main="rel. ab. Zehr.Cluster level per site") + scale_fill_manual(values=barPalette25) #rel. Abundance
rel.bar + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

  ## plot stacked bar chart per SAMPLE
rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa, main="rel. ab. Zehr.Cluster level per site") +
  scale_fill_manual(values=barPalette25) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
rel.bar  + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.sign.z.cl.fw.siteNNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.sign.z.cl.fw.siteNNP",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

#### --> stacked bar charts & heatmap @ Zehr.Subcluster level
```{r}
## prepare data for plotting
ab.Taxonomy <- ab.Taxonomy.z.subcluster
#class
tax.data2plot <- data.frame(Sample = ab.Taxonomy$Name, Freq = ab.Taxonomy$Freq, Taxa = ab.Taxonomy$Zehr.Subcluster, Zehr.Cluster = ab.Taxonomy$Zehr.Cluster, site_code = ab.Taxonomy$site_code, continent = ab.Taxonomy$continent, elevation = ab.Taxonomy$elevation, trt_NNP = ab.Taxonomy$trt_NNP)

tax.data2plot <- ddply(tax.data2plot, c("Taxa", "Sample", "site_code", "continent", "elevation", "trt_NNP"), summarise, Freq = sum(Freq)) # eintr?ge reduziert auf einen eintrag pro Class und site_code # wurde ev.schon durch merge_samples() in phyloseq gemacht

# for stacked bar charts
  ## f?gt relative abundance hinzu $Sample
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$Sample == tax.data2plot$Sample[i]])
  tax.data2plot$Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## f?gt relative abundance hinzu $trt_NNP
for (i in 1:nrow(tax.data2plot)) {
  sample.sum <- sum(tax.data2plot$Freq[tax.data2plot$trt_NNP == tax.data2plot$trt_NNP[i]])
  tax.data2plot$NNP.Rel.Ab[i] <- tax.data2plot$Freq[i] / sample.sum * 100}

  ## plot stacked bar chart per NNP
rel.bar <- qplot(x=trt_NNP,data=tax.data2plot,geom="bar",weight=NNP.Rel.Ab,fill=Taxa, main="rel. ab. Zehr.Subcluster level per site") + scale_fill_manual(values=barPalette28) #rel. Abundance
rel.bar + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

  ## plot stacked bar chart per SAMPLE
rel.bar <- qplot(x=Sample,data=tax.data2plot,geom="bar",weight=Rel.Ab,fill=Taxa, main="rel. ab. Zehr.Subcluster level per site") +
  scale_fill_manual(values=barPalette28) +
  scale_y_continuous(expand = c(0,1)) + # reduced space between bars and x-axis
  theme(axis.text.x = element_text(angle = 30, hjust = 1)) # rotating the x-axis labels
rel.bar  + facet_wrap(~site_code+trt_NNP, ncol=2, scale="free")

ggsave(paste("fierer.analyses/graphs/bar.sign.z.subcl.fw.siteNNP",Sys.Date(),"png",sep = "."),width = 10, height = 8, dpi = 300)
ggsave(paste("fierer.analyses/graphs/bar.sign.z.subcl.fw.siteNNP",Sys.Date(),"pdf",sep = "."),width = 7, height = 5, dpi = 300)
```

# getting interesting OTUs
## define target
```{r}
topN = 100
# most_abundant_taxa = sort(taxa_sums(physeq.bsc.rare.nA), TRUE)[1:topN]
# print(most_abundant_taxa)

# target = "Tabernas initial crust"
# target = "Tabernas mature crust"
# target = "Hochtor"
# target = "Svalbard"
target = "Irazu"

```

## run DESeq for target
```{r}
(physeq.bsc.subs <- subset_samples(physeq.bsc.rare.nA, Landscape%in%(target)))

# # TaIc vs. TaMc
# target = "Tabernas"
# (physeq.bsc.subs <- subset_samples(physeq.bsc.rare.nA, Samplingsite%in%(c(target))))
# (physeq.bsc.subs <- subset_samples(physeq.bsc.subs, Type%in%(c("crust"))))

# # HOc vs. SVc
# target = "HOcSVc"
# (physeq.bsc.subs <- subset_samples(physeq.bsc.rare.nA, Type%in%(c("crust"))))
# (physeq.bsc.subs <- subset_samples(physeq.bsc.subs, !Samplingsite%in%("Tabernas")))
# (physeq.bsc.subs <- subset_samples(physeq.bsc.subs, !Landscape%in%("Irazu")))
# sample_data(physeq.bsc.subs)

most_abundant_taxa = sort(taxa_sums(physeq.bsc.subs), TRUE)[1:topN]
print(most_abundant_taxa)

physeq.bsc.subs100 = prune_taxa(names(most_abundant_taxa), physeq.bsc.subs)
# physeq.bsc.Setmerged <- merge_samples(physeq.bsc.rare100, "Set")


physeq.sim <- physeq.bsc.subs100

pds <- phyloseq_to_deseq2(physeq.sim, ~ Set)

pds = DESeq(pds)


# gm_mean = function(x, na.rm=TRUE){
#   exp(sum(log(x[x > 0]), na.rm=na.rm) / length(x))
# }
# geoMeans = apply(counts(pds), 1, gm_mean)
# 
# pds = estimateSizeFactors(pds, geoMeans = geoMeans)
# 
# pds = DESeq(pds)

res <- results(pds, cooksCutoff = FALSE)
alpha=0.04
sigtab <- res[which(res$padj < alpha), ]

bmean=10 # only show OTUs with at least baseMean of 20 reads
sigtab2 <- sigtab[which(sigtab$baseMean>bmean), ]
sigtab <- sigtab2

sigtab <- cbind(as(sigtab,"data.frame"), as(tax_table(physeq.sim)[rownames(sigtab), ], "matrix"))
head(sigtab,100)
dim(sigtab)

write.table(sigtab, paste0(target,".sigtab.",alpha,".",bmean,".", topN, ".csv"), sep=";", row.names=TRUE)
# write.csv2(sigtab, paste0(target,".sigtab", topN, ".csv"))

(count.phylum <- table(sigtab$Phylum))
(count.class <- table(sigtab$Class))
(count.order <- table(sigtab$Order))
(count.family <- table(sigtab$Family))
(count.genus <- table(sigtab$Genus))
(count.cluster <- table(sigtab$Zehr.Cluster))
(count.subcluster <- table(sigtab$Zehr.Subcluster))
write.table(count.phylum, paste0(target,".sigtab.",alpha,".",bmean,".count.phylum", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.class, paste0(target,".sigtab.",alpha,".",bmean,".count.class", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.order, paste0(target,".sigtab.",alpha,".",bmean,".count.order", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.family, paste0(target,".sigtab.",alpha,".",bmean,".count.family", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.genus, paste0(target,".sigtab.",alpha,".",bmean,".count.genus", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.cluster, paste0(target,".sigtab.",alpha,".",bmean,".count.cluster", topN, ".txt"), sep="\t", row.names=FALSE)
write.table(count.subcluster, paste0(target,".sigtab.",alpha,".",bmean,".count.subcluster", topN, ".txt"), sep="\t", row.names=FALSE)
# write.csv2(count.phylum, paste0(target,".sigtab.count.phylum", topN, ".csv"))
# write.csv2(count.class, paste0(target,".sigtab.count.class", topN, ".csv"))
# write.csv2(count.order, paste0(target,".sigtab.count.order", topN, ".csv"))
# write.csv2(count.family, paste0(target,".sigtab.count.family", topN, ".csv"))
# write.csv2(count.genus, paste0(target,".sigtab.count.genus", topN, ".csv"))
# write.csv2(count.cluster, paste0(target,".sigtab.count.cluster", topN, ".csv"))
# write.csv2(count.subcluster, paste0(target,".sigtab.count.subcluster", topN, ".csv"))

otu.exp <- as.data.frame(otu_table(physeq.sim))
write.csv2(t(otu.exp), paste0(target,".otu.mat", topN, ".csv"))

# siglist <- row.names(sigtab)
# sig.comm <- otu.exp[,grep(siglist, colnames(otu.exp))]

```

####################################################################################################
################################# numeric / continuous variables ###################################
######################################## mantel / adonis ###########################################

# data preparation for mantel test and adonis
```{r}
OTU.mat.m <- OTU.mat.rare # check preparing data for plotting part 
labels.m <- as.data.frame(labels.num) # must be numeric
summary(labels.m)
labels.m.scaled <- as.data.frame(scale(labels.m)) # normalize numeric metadata
summary(labels.m.scaled)
cats <- colnames(labels.m) 
```

## mantel test
```{r}
(m1 <- mantel(vegdist(OTU.mat.m), vegdist(labels.m.scaled$MAP_WARM_Q, "euclidean"), permutations = 1000))
(m2 <- mantel(vegdist(OTU.mat.m), vegdist(labels.m.scaled$MAP_WARM_Q, "euclidean"), permutations = 1000, strata = labels$site_code))
?mantel()
```



## high throughput mantel tests
### not normalized mantel()
```{r}
d <- data.frame(matrix(ncol = 2,nrow = 0))
for (i in cats) {
  OTU.mat2test <- OTU.mat.m[!is.na(labels.m[[i]]), ]
  Labels2test <- labels.m[!is.na(labels.m[[i]]), ]
  m <- mantel(vegdist(OTU.mat2test), vegdist(Labels2test[[i]], "euclidean"), permutations = 1000)
  d[i,1] <- m$statistic
  d[i,2] <- m$signif
  print(d)
}
d$Variable <- rownames(d)
colnames(d) <- c("Mantel_statistic","p_value","Variable")
d$p_value_adj <- p.adjust(d$p_value, method = "fdr") #method="fdr", "hochberg"

(mantel_num.raw <- ggplot(data=d,aes(x=Variable,y = Mantel_statistic, label = round(p_value_adj,3), fill=Variable))+ ggtitle(plottitle)+geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=1, size=10))+geom_text(aes(y = Mantel_statistic + 0.12),check_overlap = TRUE, position = position_dodge(0.9), vjust = 0))

# ggsave("mantel_num.raw.png",width = 18, height = 12, dpi = 300)
```

### scale() - normalized
```{r}
d2 <- data.frame(matrix(ncol = 2,nrow = 0))
for (i in cats) {
  OTU.mat2test <- OTU.mat.m[!is.na(labels.m.scaled[[i]]), ]
  Labels2test <- labels.m.scaled[!is.na(labels.m.scaled[[i]]), ]
  m <- mantel(vegdist(OTU.mat2test), vegdist(Labels2test[[i]], "euclidean"), permutations = 1000)
  d2[i,1] <- m$statistic
  d2[i,2] <- m$signif
  print(d2)
}
d2$Variable <- rownames(d2)
colnames(d2) <- c("Mantel_statistic","p_value","Variable")
d2$p_value_adj <- p.adjust(d2$p_value, method = "fdr") #method="fdr", "hochberg"

(mantel_num.norm <- ggplot(data=d2,aes(x=Variable,y = Mantel_statistic, label = round(p_value_adj,3), fill=Variable))+ ggtitle(plottitle)+geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=1, size=10))+geom_text(aes(y = Mantel_statistic + 0.12),check_overlap = TRUE, position = position_dodge(0.9), vjust = 0))

# ggsave("mantel_num.norm.png",width = 18, height = 12, dpi = 300)
```

## high throughput adonis tests
### not normalized
```{r}
d3 <- data.frame(matrix(ncol=2,nrow=0))

for (i in 1:length(cats)) { 
  OTU.mat2test <- OTU.mat.m[!is.na(labels.m[[i]]), ]
  Labels2test <- labels.m[!is.na(labels.m[[i]]), ]
    form <- as.formula(paste("OTU.mat2test", cats[i], sep=" ~ "))
    a <- (adonis(form, data = Labels2test, method = "bray", permutations=999,na.rm = TRUE))
  d3[i,1] <- a$aov.tab$R2[1]
  d3[i,2] <- a$aov.tab$Pr[1]
  print(d3)
}

d3$Variable <- cats
colnames(d3) <- c("adonis_R2","p_value","Variable")
d3$p_value_adj <- p.adjust(d3$p_value, method = "fdr") #method="fdr", "hochberg"

(adonis_num.raw <- ggplot(data=d3,aes(x=Variable,y = adonis_R2, label = round(p_value_adj,3), fill=Variable))+ ggtitle(plottitle)+geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=1, size=10))+geom_text(aes(y = adonis_R2 + 0.12),check_overlap = TRUE, position = position_dodge(0.9), vjust = 0))

# ggsave("adonis_num.raw.png",width = 18, height = 12, dpi = 300)
``` 

### scale() - normalized adonis()
```{r}
d4 <- data.frame(matrix(ncol=2,nrow=0))

for (i in 1:length(cats)) { 
  OTU.mat2test <- OTU.mat.m[!is.na(labels.m.scaled[[i]]), ]
  Labels2test <- labels.m.scaled[!is.na(labels.m.scaled[[i]]), ]
    form <- as.formula(paste("OTU.mat2test", cats[i], sep=" ~ "))
    a <- (adonis(form, data = Labels2test, method = "bray", permutations=999,na.rm = TRUE))
  d4[i,1] <- a$aov.tab$R2[1]
  d4[i,2] <- a$aov.tab$Pr[1]
  print(d4)
}

d4$Variable <- cats
colnames(d4) <- c("adonis_R2","p_value","Variable")
d4$p_value_adj <- p.adjust(d4$p_value, method = "fdr") #method="fdr", "hochberg"

(adonis_num.norm <- ggplot(data=d4,aes(x=Variable,y = adonis_R2, label = round(p_value_adj,3), fill=Variable))+ ggtitle(plottitle)+geom_bar(stat="identity") + theme(axis.text.x=element_text(angle=90, vjust=1, size=10))+geom_text(aes(y = adonis_R2 + 0.12),check_overlap = TRUE, position = position_dodge(0.9), vjust = 0))

# ggsave("adonis_num.norm.png",width = 18, height = 12, dpi = 300)
```

# ------- constrained analyses using numeric / continuous values ------

# Make constrained ordination (dbRDA) using 2/3 important factors  
```{r}

Labels2test3 <- labels.m[,c("longitude", "MAP_WARM_Q", "TEMP_WET_Q")]
g1 <- cca(OTU.mat.rare ~ ., Labels2test3)
(g3 <- adonis(OTU.mat.rare ~ ., data = Labels2test3, method = "bray", permutations = 999))

autoplot(g1)

Labels <- labels.rare # Labels2test3
OTU2plot <- OTU.mat.rare
plottitle <- "constrained top3variables"

  # ## constrained plot soil_pH
  # OTU.mat2test <- OTU.mat.rare[!is.na(Labels$soil_pH), ]
  # Labels <- Labels[!is.na(Labels$soil_pH), ]
  #   dbRDA.obj <- capscale((OTU.mat2test) ~ soil_pH, data = Labels, distance = "bray")

dbRDA.obj <- capscale((OTU2plot) ~ longitude + MAP_WARM_Q + TEMP_WET_Q, data = Labels, distance = "bray")
# dbRDA.obj <- capscale((OTU2plot) ~ MAP_WARM_Q + TEMP_WET_Q, data = Labels, distance = "bray")
# dbRDA.obj <- capscale((OTU2plot) ~ longitude + MAP_WARM_Q + TEMP_WET_Q, data = Labels, distance = "bray")
# autoplot(dbRDA.obj)

dbRDA.scores <- scores(dbRDA.obj)
explained <- eigenvals( dbRDA.obj )/sum( eigenvals( dbRDA.obj ) ) * 100
explained <- format( round( explained, 1 ), nsmall = 1 )

dbRDA.df <- data.frame(id = row.names(dbRDA.scores$sites), dbRDA.scores$sites, sampleID = Labels$sampleID, Names = Labels$Names, conc = Labels$conc, site_code = Labels$site_code, continent = Labels$continent, country = Labels$country, region = Labels$region, trt = Labels$trt, nutrient_trt = Labels$nutrient_trt, soil_pH = Labels$soil_pH, trt_yn=Labels$trt_yn, trt_NNP = Labels$trt_NNP, trt_NPP = Labels$trt_NPP)

colnames(dbRDA.df)[c(2,3)] <- c("PC1", "PC2")
p <- PlotOrd(dbRDA.df , components = c("PC1", "PC2"), groups = "soil_pH", treatment = "continent", variance = TRUE, connect = FALSE)

p + scale_size_manual(values = c(4, 8)) + ggtitle(plottitle) +
  xlab(paste("dbRDA 1 (", explained[1], "%)", sep = "")) + 
  ylab(paste("dbRDA 2 (", explained[2], "%)", sep = ""))

# ggsave("dbRDA_autoplot.png",width = 10, height = 8, dpi = 300)
```

